probe-rs/src/architecture/riscv/communication_interface.rs

```diff
@@ -182,12 +206,13 @@ pub struct RiscvCommunicationInterface {
 }
 
 impl<'probe> RiscvCommunicationInterface {
-    pub fn new(probe: Box<dyn JTAGAccess>) -> Result<Self, DebugProbeError> {
+    pub fn new(probe: Box<dyn JTAGAccess>) -> Result<Self, (Box<dyn JTAGAccess>, DebugProbeError)> {
```

Comment:
Is it Rust-idiomatic to return a Result from new()? Or should this then rather be a try_new or try_from?

Reply:
Not sure if it is idiomatic, but we do it like this in more places.